### DISCUSSION TITLE
MySQL - Data - Sources - Security

### DISCUSSION THREAD

**Discussion by mmalessa80** [Budibase/budibase#6239] [Ideas]
I have literally just started exploring Budibase as a potential CRUD app solution and have found the building interface to be superb. My background primarily is simple free standing applications in Google Apps Script, then a similar application builder that was Google App Maker before it was deprecated about 2 years ago, with my most current experience in AppSheet.

With that being said, there appears to be a possible shortcoming (or maybe I am just missing how to implement this) by having the ability to use security filters in MySQL tables under the data sources section. Although there is a 'filter' available, this filter does not actually appear to offer any security to the data as to what is exposed to the Budibase application and how this in turn prevents someone from the client to access all data that is synonymous with the SQL tables. Please correct me if I am wrong in my assertion.

My idea in this case consists of a couple of pieces. Lets say that my particular application does not have the need to access all tables within my SQL database, so when linking a MySQL database have a multi-select of the available tables and then only select the tables you wish to expose in the application. Furthermore, once tables are available have the ability to further filter (via a query possibly) what data is available server side. In this instance I think it would be important to still have access to Current User bindings on the server side since often data is filtered based on the logged in user.

Although it appears some of this is available by using Custom Queries along with bindings, it seems that those bindings originate from the client and could potentially be intercepted to alter what the query might return. Again, please correct me if I am wrong with this assumption.

**Comment by melohagan** [comment]
Hi @mmalessa80 

Thank you for exploring Budibase!

There is no granular control for data in the data tab, at the moment. This issue may be related: https://github.com/Budibase/budibase/issues/3421

In future hopefully we will have the option to include both a production and dev connection for external datasources. This would mean that only dev data would be displayed within the builder, and production data would be kept segregated, and only used in the published app.
